### ISSUE TITLE
Tech rework

### ISSUE THREAD

**Issue by Helliaca** [Helliaca/HoI4-LotrMod#12]
It comes down roughly to:

- [x] Changing Artillery to Bows/Ballistas:
    - [x] Change tech icons
    - [x] Adapt stats
- [x] Changing Tanks to Cavalry:
    - [x] Change tech icons
    - [x] Adapt stats
- [x] Do something with planes (Eagles/Nazgul probably)
- [x] Add custom units
    - [x] Catapults
    - [x] Trebuchets
    - [x] Mumakil
    - [ ] Ents
    - [ ] Mithril units

I'm probably missing plenty of things, but to get a rough idea of what milestones we should strive towards.
Once everything regarding tech makes more sense, we should also pass over all the focus trees and re-name (+ icons) all the focuses called "Heavy Tank Focus" or similar.

For a project much further into the future: we can also look into including more adequate 3d models for all the custom units. Cavalry should be easy. Trebuchets and Catapults shouldn't be too hard, Mumakil and Ents might be significantly more difficult.

**Comment by Ryless** [comment]
I was thinking of keeping the mithril aspect simple and just adding new mithril equipment which would be unlocked with the  infantry equipments with better stats and higher production cost. I have done this already really so it would be quite simple but let me know if you want to make them separate units and techs. The issue with this stuff is I'm not exactly sure how to prevent nations from building mithril equipment even though they don't have any. In Hoi4 you simply get a production penalty for lacking resources if you don't have steel for example but can still build the items. I don't like this feature when it comes to mithril equipment, if possible I'd like it not to be possible at all to build mithril equipment if you don't have any mithril. I haven't found a great way to do this yet. 

When it comes to stats for cavalry I believe the first thing we should do is maybe increase the speed a tiny bit and make the hardness/softness more realistic since they are horses and not metal machines anymore. I don't know what other stats we would change for that necessarily. 
For bows and ballistae, I'm not sure what we would modify.


I don't love the plane situation since canon wise really only Mordor should have anything and eagles are their own beings that help out whenever, but that would be dumb for the mod obviously. The air mechanics are really important for the game in general so messing with it too much is probably a bad idea. The most simple thing we could do is change the names and icons to eagles and fell beasts respectively so it's probably the smartest thing to do. Even making all the planes fighters capable of air support, interception, ground support, etc, since they didn't have designated roles, seems too extreme.

I'll work on adding the custom units. For the techs, I was thinking to have trebuchets split off of catapults at some point. I think maybe having siege ladders behind techs that connect to siege tower tech would also make sense. The techs for the country-specific units I was thinking of putting on the cavalry page off to the side a little bit and adding a tooltip explaining who gets them.

The more I think about adding a siege tab and separating cavalry and siege machines, the more I want to do it. I feel like we should probably have more than catapults/trebs and ladders/siege towers, and probably should have more cavalry as well, if we were to do that though. So I'll hold off on actually doing it.

Perhaps, we could add another tier of light cav and medium cav to have more things to research since we've removed a few. This would be pretty easy to do but let me know what you think. It might not be necessary.

As for 3d models, I have no experience with anything like that so I'm not sure how much help I'd be but obviously, I think that would be a great thing to add in the future. For light cavalry, The base cavalry model would work okay right away.

**Comment by Ryless** [comment]
> I haven't found a great way to do this yet.

can_be_produced = {
		ROOT = {has_country_flag = can_build_mithril_equip}
		}
In the equipment file combined with events assigning the country flag when it should be, seems to have worked. Hypothetically I shouldn't have to use events to do this, I feel like I should be able to just put the triggers for the event in the scope above but that threw an error about dynamic tokens which doesn't make sense to me since the resource variables are inherently dynamic.

Whatever though, It seems to work and update in-game when a country trades for mithril.

**Comment by Ryless** [comment]
How do you feel about lowering the population across the board? Lowering state population and then lowering manpower needed for units wouldn't be too much work and I know it has been suggested before.

**Comment by Helliaca** [comment]
Yeah, I think its a good idea and makes sense, especially with reworked the techs.

The only real downside I can think of, is that players won't have a "frame of reference" when it comes to manpower numbers. So a focus that gives, for instance, 2.5k manpower won't really say much as the players are used to ww2 numbers. Even so, I still think its a good idea.

I think a good factor of population would be something like 0.05 or 0.06. Lore-wise Gondor has a population of about 3.5million. In the mod they currently have 63million. (0.05 * 63 = 3.15).

There is also a monthly population growth number. But afaik it is percentage-based (0.15% monthly or something), so I guess that shouldn't be a problem either.

**Comment by Helliaca** [comment]
Btw, and completely unrelated to the above:

I've been looking at some old stuff of mine and found something that might work quite well in this mod.

Take a look at this tech tree from an unreleased, sci-fi type mod I made:
![techtree](https://user-images.githubusercontent.com/18307020/116283341-12f15d00-a78c-11eb-9e76-8ef39cc4719f.jpg)

If you look in particular at the left side (infantry equipment), you will see that I divided it up and made the player choose between "sophisticated" and "affordable"/regular infantry equipment. The sophisticated equipment has far higher reliability, somewhat better numbers but costs more and requires "Silicon" (a resource) to produce. The "highly sophisitcated" equipment even has a tiny bit of anti-air, while the "mass-producible" is even cheaper than tier0 (starting) equipment but still has okay-ish numbers.

Concerning the LOTR mod: Almost all equipment that requires steel also requires wood. This makes wood and steel almost into a single, interchangeable resource. 

I think it might be a good idea to also divide our equipment in a similar manner. "Affordable"/cheap equipment would essentially be things like a crude machete. Essentially this:
![](https://i.stack.imgur.com/Q1jOi.jpg)
And would not require wood at all, but also we less effective. A natural choice for mordor (which would have less wood) or Isengard if they decide *not* to burn the Entwood. These would instead use the precious little wood they have for artillery (bows).

Having large deposits of wood (or burning the entwood) would give a nation the luxury of producing "sophisticated" equipment. This would essentially be Halberds, Spears, shields and the likes. They need wood to produce and slightly more expensive, but also generally better.

Maybe Mithril equipment would have a place as "highly sophisitcated" as shown in the tech-tree screenshot.

"Mass-producible" equipment *could* be made of *only wood* and be quite bad, maybe for meme-builds :D

I think this dichotomy would make the wood-steel balance a little more interesting as well as give the player additional options/choices. A good thing, since we have already diminished these by the removal of regular cavalry and super-heavies.

Let me know what you think.

**Comment by Ryless** [comment]
This is an interesting idea. I have been having some issues with the whole mithril idea and so this could potentially be a solution to that but I'm not entirely sure it would. I can't find a way to make the AI mix between the two types of infantry equipment and I don't know if that is even something that I can manipulate them into doing. They will use like 15 factories on mithril equipment even if they only have 1 mithril or won't use it at all. I could be missing something but if I'm not I might have to make it a different tech and unit entirely. This isn't super ideal to me but could work.

I also did have thoughts about the equipment balance in general but thought we could do that by focus tree decisions, this didn't include the wood-steel balance which is definitely a fair suggestion. Your idea would probably work better but would take more work. Since you already have something in place it may not be difficult to implement.

Also, I've added all of the custom units with placeholders to my fork called TechUpdate if you want to look at anything in there. The stats are very raw still and not tested, some are not really changed at all. It also will show how many techs are now in the cavalry tab that makes it quite cluttered and could benefit from a siege tab but as I said before I feel like we'd have to add more things to cavalry and siege tabs to put them in a good spot. I've assigned the countries to their respective techs and fixed the OOBs to include the new units. As far as the Custom units, they are in the game but probably not balanced either so you can check those off if you'd like.

**Comment by Helliaca** [comment]
> I can't find a way to make the AI mix between the two types of infantry equipment.

Afaik the AI will always just produce the "newest" equipment. Your separate tech and unit solution might work. Otherwise, maybe it is best if we leave all the mithril stuff as something exclusive to players.

> Since you already have something in place it may not be difficult to implement.

I tried just dropping the mod files right into the lotr mod, but the files are badly outdated and kind of broken. There is *some* kind of result on the [tech-tree-experiment](https://github.com/Helliaca/HoI4-LotrMod/tree/tech-tree-experiment) branch. It might be easier to just do it from scratch?

**Comment by Ryless** [comment]
I've updated https://github.com/Ryless/HoI4-LotrMod/tree/TechUpdates with the new manpower numbers at 6% of the original numbers. I changed all the manpower references I could find to 6% as well.  I imagine some further balancing could be useful but I'm not sure.

The only things that we really have to do are: 

- Adapt stats better
- Add custom icons
- Add division icons
- Organize tech trees
- Change Planes (For this one I imagine just naming them Eagle (CAS), Eagle (Fighter) is the preferred method for now?)

**Comment by Helliaca** [comment]
I took a quick glance at it and here's a few small things I noticed:

- I can use both "Crossbowmen" and "Crossbow Marksmen" in my divisions, where "Crossbow Marksmen" use bows. I'm guessing this is an old remnant from me which still needs to be removed.
- I'm guessing you haven't touched the stats that much yet, but its a bit weird how "Light Cavalry" has more armor than "Chariot"
- "M. Cavalry" (which I'm guessing stands for "Medium Cavalry", should probably just be called "Heavy Cavalry". It makes no sense to have a "medium" if there is nothing above it (in terms of cavalry).

Overall I do really like your tank tech tree though. As you said, we still do need icons for the tech and the divisions.

**Comment by Ryless** [comment]
I fixed the things you mentioned above. There is still a ton of balancing to do for pretty much everything. I am going to work on the AI strategy/templates stuff now to try to get more fitting army compositions. Do you feel like a division limiter is needed in this mod? Generally, it's a good idea to have but I don't know if it's necessary for this mod. Also, I think I'm going to add another heavy and light cav tech so that a few nations could start with knights. Since the gameplay goes pretty quick I'm also going to rebalance the tech dates so that most of the tech is researchable, without penalities, earlier than 3025?? or something similar.

**Comment by Helliaca** [comment]
> Do you feel like a division limiter is needed in this mod? 

I don't think it desperately *needs* it. But it might be good for late game. I personally haven't had too much division spam issues with this mod.

> Another solution (...) is making wood the primary resource for most nations, like steel is in the base game. Then having steel be more like chromium

I generally like solution I described [here](https://github.com/Helliaca/HoI4-LotrMod/issues/12#issuecomment-827780325) more, but that doesn't mean they are mutually exclusive.
I feel like the primary bottlenecks for producing equipment types should be:
- Ships -> A *ton* of wood, maybe little steel for big ones
- Siege machines -> Also a whole lot of wood. A little leather as well
- Artillery/Bows -> A little wood, string should be the primary limiting factor for most nations. No steel cost at all. (Ballistas cost more wood than regular bows)
- Cavalry -> Primary limiting factor should be leather, heavy cav takes some steel as well. Cav in general should have really long training time and high food consumption.
- Chariot/War wagons -> cost the same as cavalry but with some extra wood.
- Planes -> ???
- Supp Equipment -> Some combination of steel/wood/leather. I have no preference here.
- Infantry equipment: Any regular infantry equipment should *definitely* cost steel. I feel like the wood cost is not necessary. If we think of the equipment as being "sword + armor", steel about covers that. I do, however, like the idea of giving the player the choice of either building good infantry equipment that costs wood, or worse infantry equipment that costs no wood, hence the tech tree I posted above.

I'm not entirely sure what you mean by making wood the *primary* resource. I feel like apart from siege machines and ships there isn't a single piece of equipment that makes sense to cost wood but *no* steel.
